scripts/pseudo_hadd.py

```diff
@@ -0,0 +1,144 @@
+                string = key2.ReadObj().Clone()
+                f.Close()
+                target.cd()
+                string.Write(key.GetName())
+
+        elif isinstance(obj, ROOT.TH1):
+            # descendant of TH1 -> merge it
+            print("[INFO] Merging histogram {}/{}".format(path, obj.GetName()))
+            h1 = obj.Clone()
+            # loop over all source files and add the content of the
+            # corresponding histogram to the one pointed to by "h1"
+            for fname in sourcefiles[1:]:
+                f = ROOT.TFile.Open(fname, "READ")
+                # make sure we are at the correct directory level by cd'ing to path
+                f.cd(path)
+                key2 = ROOT.gDirectory.GetListOfKeys().FindObject(h1.GetName())
+                if key2:
+                    h2 = key.ReadObj()
+                    h1.Add(h2)
+                    del h2
+                f.Close()
+            target.cd()
+            h1.Write(key.GetName())
+        
+        elif isinstance(obj, ROOT.TTree):
+            tree_name = obj.GetName()
+            tchain = ROOT.TChain(tree_name)
+            full_tree_path = "{}/{}".format(path, tree_name)
+            print("[INFO] Linking trees in {} files under '{}'...".format(len(sourcefiles), full_tree_path))
+            for fname in sourcefiles:
+                if not tchain.AddFile(fname, ROOT.TTree.kMaxEntries, full_tree_path):
+                    raise RuntimeError('[ERROR] Could not connect tree {} in file {} to TChain!'.format(full_tree_path, fname))
+                # test the links
+                if check:
+                    num_entries = tchain.GetEntries()
+                    print("[INFO] Linking successful. Combined TChain yielded {} entries.".format(num_entries))
+            target.cd()
+            tchain.Write()
+        
+        elif isinstance(obj, ROOT.TDirectory):
+            # it's a subdirectory
+            print("[INFO] Found subdirectory " + obj.GetName())
+            
+            # create a new subdir of same name and title in the target file
+            target.cd()
+            newdir = target.mkdir(obj.GetName(), obj.GetTitle())
+            
+            # newdir is now the starting point of another round of merging
+            # newdir still knows its depth within the target file via
+            # GetPath(), so we can still figure out where we are in the recursion
+            MergeRootFiles(newdir, sourcefiles, check=check)
+            
+        else:
+            print("[WARNING] Unknown object type, name: " + obj.GetName() + " title: " + obj.GetTitle())
+    target.SaveSelf(ROOT.kTRUE)
+    ROOT.TH1.AddDirectory(status)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-o", "--output", help="output root file", default="out.root")
+    parser.add_argument("-i", "--input", help="XRootD path e.g. root://cmsxrootd-kit.gridka.de//store/user/cverstege/Excalibur/dataA/*.root", type=str)
```

Comment:
Since you try to convert SRM paths into XRootD paths, shouldn't this option also reflect this functionality in the help message?

Reply:
I think I will remove the functionality. This was copied from the xrootd hadd script.

Reply:
That seems reasonable, since this feature would be pointless for SRM.